utils/remote_control/navigator_emergency_control/nodes/nodes/navigator_emergency.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+
+
+from remote_control_lib import RemoteControl
+import rospy
+from sensor_msgs.msg import Joy
+
+
+__maintainer__ = "David Zobel"
+__email__ = "zobeldavid@ufl.edu"
+__copyright__ = "Copyright 2016, MIL"
+__license__ = "MIT"
+
+
+rospy.init_node("emergency")
+
+
+class Joystick(object):
+
+    def __init__(self):
+        self.force_scale = rospy.get_param("~force_scale", 600)
+        self.torque_scale = rospy.get_param("~torque_scale", 500)
+
+        self.remote = RemoteControl("emergency", "/wrench/emergency")
+        rospy.Subscriber("joy_emergency", Joy, self.joy_recieved)
+
+        self.active = False
+        self.reset()
+
+    def reset(self):
+        '''
+        Used to reset the state of the controller. Sometimes when it
+        disconnects then comes back online, the settings are all out of whack.
+        '''
+        self.last_kill = False
+        self.last_station_hold_state = False
+        self.last_auto_control = False
+        self.last_emergency_control = False
+        self.last_shooter_cancel = False
+        self.last_change_mode = False
+
+        self.start_count = 0
+        self.last_joy = None
+        self.active = False
+
+        self.remote.clear_wrench()
+
+    def check_for_timeout(self, joy):
```

Comment:
Well, my timeout function is supposed to only check if the hardware disconnects or associated node dies and stops publishing. I still want the controller to timeout after 15 minutes of no change.

The difference is, if the node dies, /joy_emergency stops publishing completely, and it should use my timeout to see that. If the controller just isn't touched, and the node is still alive, /joy_emergency is still publishing zeros, and shouldn't activate my timeout. It should still activate your timeout after 15 minutes if the controller doesn't change state.

On closer inspection, it looks like my function didn't behave the way I just described, so I reworked it to look at a new variable last_time, which is the last time of a received /joy_emergency. Also renamed it die_check.